exchanges/stream/websocket_connection.go

```diff
@@ -293,7 +292,11 @@ func (w *WebsocketConnection) Shutdown() error {
 		return nil
 	}
 	w.setConnectedStatus(false)
-	return w.Connection.UnderlyingConn().Close()
+	w.writeControl.Lock()
+	defer w.writeControl.Unlock()
+	defer w.Connection.NetConn().Close() // Ungraceful close as backup
+	// Gracefully close the connection
+	return w.Connection.WriteMessage(websocket.CloseMessage, websocket.FormatCloseMessage(websocket.CloseNormalClosure, ""))
```

Comment:
TL;DR: Just close it.

I'm afraid these are meaningless together.
If you're not going to wait for the response, you haven't closed the connection gracefully.
From the perspective of the server they saw:
"_Hey, We're going to close the webso..._" <slam>🚪 💨 

I'm assuming you read [this discussion](https://github.com/gorilla/websocket/issues/448) as the starting point for this, but most notably the last two comments seem relevant.

I think all servers are fine with you just closing it. They don't expect any better.
There's no information gained by signalling you're going to shut down.
It's not like they'd say "we had this order message for you, and we were hoarding it, but here, have it now".

Reply:
That's fair I didn't read the discussion just saw an abnormal close and thought to change it out. I guess I will just close it then 🤷